exatomic/core/universe.py

```diff
@@ -349,3 +349,59 @@ def basis_function_contributions(universe, mo, mocoefs='coef',
         return joined
     else:
         raise NotImplementedError("not clever enough for that.")
+
+
+def to_cube(universe,file_name='output',field_number=0)
+    """
+    Provided a universe with atom and field attributes, write to a file 
+    in cube format for a single 3D scalar field.
+    
+    Warning:
+        This function is not fully tested yet!
+
+    Args:
+        universe (:class:`~exatomic.core.universe.Universe`): a universe
+        file_name (str): name of the output file without file extension
+        field_number (int): number of the single field starting with 0
+
+    Returns:
+        None
+    """
+    uni = universe
+    n = field_number
+    if os.path.isfile(file_name+'.cube'):
+        raise FileExistsError('File '+file_name+'.cube '+'exists.')
+    else:
+        f_out = open(file_name+'.cube','w')
+    f_out.write('EXATOMIC GENERATED FIELD VALUES TO CUBE FILE\n')
+    f_out.write('OUTER LOOP: X, MIDDLE LOOP: Y, INNER LOOP: Z\n')
+    f_out.write('{:4}{:12.6f}{:12.6f}{:12.6f}\n'.format(len(uni.atom),
+                                                 uni.field['ox'][n], uni.field['oy'][n], 
+                                                 uni.field['oz'][n]))
+    NX = uni.field['nx'][n]
+    f_out.write('{:4}{:12.6f}{:12.6f}{:12.6f}\n'.format(NX,
+                                                 uni.field['dxi'][n], uni.field['dxj'][n], 
+                                                 uni.field['dxk'][n]))
+    NY = uni.field['ny'][n]
+    f_out.write('{:4}{:12.6f}{:12.6f}{:12.6f}\n'.format(NY,
+                                                 uni.field['dyi'][n], uni.field['dyj'][n], 
+                                                 uni.field['dyk'][n]))
+    NZ = uni.field['nz'][n]
+    f_out.write('{:4}{:12.6f}{:12.6f}{:12.6f}\n'.format(NZ,
+                                                 uni.field['dzi'][n], uni.field['dzj'][n], 
+                                                 uni.field['dzk'][n]))
+    for index, row in uni.atom.iterrows():
+        a = row['Z']
+        #Nuclear charge on atom a, this ought to deviate from the atomic number such as when
+        #an ECP is used, although I couldn't find an entry for this so assume a is equal to
+        #atomic number Z
+        f_out.write('{:4}{:12.6f}{:12.6f}{:12.6f}{:12.6f}\n'.format(row['Z'],
+                                                                    a, row['x'],
+                                                                    row['y'], row['z']))
+    for i in range(NX):
+        for j in range(NY):
+            for k in range(NZ):
+                f_out.write(f'{uni.field.field_values[n][i*NY*NZ+j*NZ+k]:13E} ')
+                if k%6 == 5:
+                    f_out.write('\n')
+            f_out.write('\n')
```

Comment:
To flush the f_out buffer I would include a `f_out.close()` at the end of this function.